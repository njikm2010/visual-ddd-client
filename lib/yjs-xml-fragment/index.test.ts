import { Doc, XmlElement, XmlText } from 'yjs';
import { xmlFragmentToJSON, buildXmlFragmentFromJSON } from './index';
import data from './index.data.json';

test('xmlFragmentToJSON', () => {
  const doc = new Doc();

  const fragment = doc.getXmlFragment('test');

  // 普通文本
  fragment.push([new XmlText('hello'), new XmlText('world')]);

  // 格式化文本
  const formatText = new XmlText();
  formatText.insert(0, 'format text');
  formatText.format(0, 5, { bold: true });

  fragment.push([formatText]);

  const element = new XmlElement('div');
  element.setAttribute('class', 'test');
  element.setAttribute('id', 'test');

  fragment.push([element]);

  expect(xmlFragmentToJSON(fragment)).toEqual([
    {
      deltas: [
        {
          insert: 'hello',
        },
      ],
      type: 'text',
    },
    {
      deltas: [
        {
          insert: 'world',
        },
      ],
      type: 'text',
    },
    {
      deltas: [
        {
          attributes: {
            bold: true,
          },
          insert: 'forma',
        },
        {
          insert: 't text',
        },
      ],
      type: 'text',
    },
    {
      attrs: {
        class: 'test',
        id: 'test',
      },
      content: undefined,
      tag: 'div',
      type: 'element',
    },
  ]);
});

test('buildXmlFragmentFromJSON', () => {
  const doc = new Doc();

  const fragment = doc.getXmlFragment('test');
  buildXmlFragmentFromJSON(fragment, data as any);

  expect(fragment).toMatchSnapshot();
});
