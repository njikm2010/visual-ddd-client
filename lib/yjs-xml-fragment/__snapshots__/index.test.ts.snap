// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildXmlFragmentFromJSON 1`] = `"<heading level="2" textAlign="left">统一语言产品文档</heading><paragraph textAlign="left"><link class="null" href="http://www.wakedata.net/pages/viewpage.action?pageId=27711055" target="_blank">http://www.wakedata.net/pages/viewpage.action?pageId=27711055</link></paragraph><paragraph textAlign="left"></paragraph><paragraph textAlign="left"></paragraph><paragraph textAlign="left"><bold>原生地址：</bold><bold><link class="null" href="http://www.wakedata.net/pages/viewpage.action?pageId=34996708" target="_blank">http://www.wakedata.net/pages/viewpage.action?pageId=34996708</link></bold></paragraph><paragraph textAlign="left"></paragraph><paragraph textAlign="left"></paragraph><heading level="2" textAlign="left">约束和限制</heading><blockquote><paragraph textAlign="left">【可选】<hardbreak></hardbreak>列出明确的约束和限制，或者说要达到的目标，如性能指标，原则上这些指标需要优于技术委员会架构规范要求的指标，常见的约束和限制有：</paragraph><orderedlist start="1"><listitem><paragraph textAlign="left">质量，扩展性要求</paragraph></listitem><listitem><paragraph textAlign="left">性能要求</paragraph></listitem><listitem><paragraph textAlign="left">安全要求</paragraph></listitem><listitem><paragraph textAlign="left">项目时间要求</paragraph></listitem><listitem><paragraph textAlign="left">硬件，软件选型要求</paragraph></listitem><listitem><paragraph textAlign="left">成本的要求</paragraph></listitem><listitem><paragraph textAlign="left">监管方的监管要求</paragraph></listitem></orderedlist><paragraph textAlign="left"><bold>Notice：</bold>约束限制越多越好！</paragraph><paragraph textAlign="left"></paragraph></blockquote><paragraph textAlign="left"></paragraph><paragraph textAlign="left"></paragraph><heading level="2" textAlign="left">质量设计</heading><blockquote><paragraph textAlign="left">【可选】</paragraph><paragraph textAlign="left">描述指令相关的设计，包括可观测性，可维护性，可测试性，成本设计等，比如单元测试需要达到的百分比，原则上这些设计指标需要优于技术委员会架构规范约定的指标</paragraph></blockquote><paragraph textAlign="left"></paragraph><paragraph textAlign="left"></paragraph><paragraph textAlign="left"></paragraph><paragraph textAlign="left"></paragraph><heading level="2" textAlign="left">演进规划</heading><blockquote><paragraph textAlign="left">【可选】</paragraph><paragraph textAlign="left">可以是演进规划， 也可以是产品规划，描述每一个里程碑或者版本计划</paragraph></blockquote><paragraph textAlign="left"></paragraph><paragraph textAlign="left"></paragraph><paragraph textAlign="left"></paragraph><paragraph textAlign="left"></paragraph><heading level="2" textAlign="left">其他</heading><paragraph textAlign="left"></paragraph><blockquote><paragraph textAlign="left">【可选】</paragraph><paragraph textAlign="left">如果复杂的，有架构设计等等的，可以再补充，或其他限制、注意事项等<hardbreak></hardbreak>如： 部署设计，风险评估，应对策略等</paragraph></blockquote><paragraph textAlign="left"></paragraph><paragraph textAlign="left"><hardbreak></hardbreak></paragraph>"`;
